crypto/pkcs11key/key.go

```diff
@@ -166,12 +204,18 @@ func (ps *Key) setup(privateKeyLabel string) (err error) {
 	}
 	ps.privateKeyHandle = privateKeyHandle
 
-	publicKey, err := getPublicKey(ps.module, session, privateKeyHandle)
+	publicKey, err := ps.getPublicKey(ps.module, session, privateKeyHandle)
 	if err != nil {
 		ps.module.CloseSession(session)
 		return
 	}
-	ps.publicKey = publicKey
+	if ps.keyType == EC {
+		t := publicKey.(ecdsa.PublicKey)
```

Comment:
I'll try to change to pointers and see if the switch on the publicKey works too.